packages/babel-core/src/config/files/configuration.js

```diff
@@ -168,7 +168,7 @@ const readConfigJS = makeStrongCache(function* readConfigJS(
   }>,
 ): Handler<ConfigFile | null> {
   if (!fs.exists.sync(filepath)) {
-    cache.forever();
+    cache.never();
```

Comment:
If we disable the cache, Babel will be constantly searching js config files (`babel.config.(cjs|mjs|js)` and `.babelrc.(cjs|mjs|js)` alongside the folder structures). I doubt the redundant disk IO here is worthy.

I tend to leave as-is until we have a better solution. As @devongovett mentioned before, it is difficult to invalidate JS configs. If one renames `babel.config.json` to `babel.config.js`, they are likely to make some changes, too. It is confusing if adding config invalidate the cache but not modifying.